os/fs/smartfs: Update jSmartFs overwrite to ensure file data consistency

amogh-samsung opened the pull request:
Sample PR, not for official review

These commits will upgrade jSmartFs overwrite from a file system integrity level to a file data consistency level

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

thapav committed:
arch/arm/src: Add IRQ debug logs during assertion

This patch:
1) Adds a global variable g_irq_nums to track the current and previous
irq number
2) Adds debug logs for IRQ during assert
3) Refactors debug logs during assert
4) Adds logs for different scenarios in case of assert such as IRQ,
Nested IRQ, workqueue and normal thread

for armv7-m and armv8-m.

Sample logs are as follows:
```
**************************************************
INFO: Read another app space: 0x2100028
up_hardfault: PANIC!!! Hard fault: 40000000
up_assert: Assertion failed at file:armv8-m/up_hardfault.c line: 204 task: tash
up_dumpstate: sp:     0218562c
up_dumpstate: Code asserted in IRQ state!
up_dumpstate: IRQ num: 4
up_dumpstate: IRQ handler: 0e008fd5
up_dumpstate: SP is not in IRQ stack, but it lies in:
up_dumpstate: User stack:
up_dumpstate: stack base: 02185878
up_dumpstate: stack size: 00000fec
up_dumpstate: stack used: 00000334
up_stackdump: 02185620: xxxxxxxx xxxxxxxx xxxxxxxx 000000cc 1007222c 00000001 00000330 00000018
up_stackdump: 02185640: 0e006a1b 00000003 0e00715d 0e00679d 0000002c 1002c1ec 1002c1ec 1002a9d0
up_stackdump: 02185660: 00000003 0218569c 0218570c 00000001 00000330 00000018 0e00940d 0e0093e5
up_stackdump: 02185680: 10012169 1002c1ec 1002a9d0 00000004 0218570c 00000000 10011e4b 021856f0
up_stackdump: 021856a0: 000000e0 1002c1ec 1002a9d0 00000004 0218570c 00000000 00000001 00000330
up_stackdump: 021856c0: 00000018 ffffffb0 02185740 00000000 00000004 0218570c 00000000 00000000
up_stackdump: 021856e0: 00000000 10012169 0e008fd8 41000204 10011e4b 00000004 00000000 00000022
up_stackdump: 02185700: 0217f348 0217f348 10011e4b 02185760 000000e0 00000004 00000000 00000022
up_stackdump: 02185720: 0217f348 0217f348 00000001 00000330 00000018 ffffffbc 00000000 02184890
up_stackdump: 02185740: 00000000 00000000 00000000 02100028 10026fa5 0220040b 02205612 61000000
up_stackdump: 02185760: 00000002 02100028 00000000 021857a8 00000000 022021cf 021857a8 00000001
up_stackdump: 02185780: 00000000 021858a8 00000001 00000000 00000001 00000000 00000000 00000000
up_stackdump: 021857a0: 00000000 02202829 021858a0 00000000 00000000 00000000 00000000 ffffffbc
up_stackdump: 021857c0: 00000000 02184890 00000001 100287a0 00000001 00000000 0e001eb3 02202679
up_stackdump: 021857e0: 0220030a 21000000 021858a0 02202679 02185820 02185848 000000e0 00000000
up_stackdump: 02185800: 0217fcd4 00000008 00000000 00000008 00000001 00000000 00000009 00000008
up_stackdump: 02185820: 00000006 00000000 00000007 00000003 0217fcd4 021858a0 022591cc 00000000
up_stackdump: 02185840: 00000000 00000000 00000000 022028a3 0218587c 02202831 00000001 0218587c
up_stackdump: 02185860: 00000000 0220043d 0218587c 0e001c83 00000000 00000000 deadbeef 02185884
up_registerdump: R0: 00000004 0218570c 00000000 00000000 1002c1ec 1002a9d0 00000004 0218570c
up_registerdump: R8: 00000000 00000001 00000330 00000018 00000000 021856f0 10012169 0e008fd8
up_registerdump: xPSR: 41000204 BASEPRI: 000000e0 CONTROL: 00000001
up_registerdump: EXC_RETURN: ffffffb0
task_show_alivetask_list: *******************************************
task_show_alivetask_list: List of all tasks in the system:
task_show_alivetask_list: *******************************************
task_show_alivetask_list:                            NAME |   PID |  PRI |    USED / TOTAL STACK | STACK ALLOC ADDR | TCB ADDR
task_show_alivetask_list: ---------------------------------------------------------------------------------------------------
task_taskdump:                       Idle Task |     0 |    0 |     668 /    1024 |       0x10004bfc | 0x100293b0
task_taskdump:                          hpwork |     1 |  201 |     268 /    2028 |       0x1005eda0 | 0x1005e640
task_taskdump:                          lpwork |     2 |   50 |     268 /    2028 |       0x1005fd10 | 0x1005f5b0
task_taskdump:                     LWIP_TCP/IP |     3 |  105 |     360 /    4068 |       0x1006c330 | 0x1006bbd0
task_taskdump:            netmgr_event_handler |     4 |  100 |     212 /    2012 |       0x1006daa0 | 0x1006d340
task_taskdump:                  binary_manager |     5 |  203 |     760 /    8164 |       0x1006ea80 | 0x1006e320
task_taskdump:               bm_faultmsgsender |     6 |  205 |     220 /     988 |       0x100711f0 | 0x10070a90
task_taskdump:                            app1 |     8 |  220 |     412 /    4068 |        0x21003d0 | 0x10072f10
task_taskdump:                            app2 |    10 |  180 |     632 /    8164 |        0x2181860 | 0x100717a0
task_taskdump:                           uwork |    11 |  100 |     292 /    2028 |        0x2183dd0 | 0x10071c50
task_taskdump:                            tash |    12 |  125 |     892 /    4076 |        0x2184890 | 0x10072100
task_taskdump:                wifi msg handler |    13 |  100 |     404 /    4060 |        0x2185f10 | 0x100725b0
mpu_show_regioninfo: ********************************************************************************************
mpu_show_regioninfo: REGION_NO.         BASE_ADDRESS    SIZE            STATUS          ACCESS (P/U)    EXECUTE
mpu_show_regioninfo: ********************************************************************************************
mpu_show_regioninfo: 0                  0x10042D40      0x157A0         ENABLED         RW/RW           X
mpu_show_regioninfo: 1                  0x2000000       0x100020        ENABLED         RW/RW           X
mpu_show_regioninfo: 2                  0x21FE4C0       0x58E20         ENABLED         RO/RO           X
mpu_show_regioninfo: 3                  0x22572E0       0x21EA0         ENABLED         RO/RO           XN
mpu_show_regioninfo: 4                  0x217F340       0x7F180         ENABLED         RW/RW           XN
mpu_show_regioninfo: 5                  0x0             0x20            DISABLED        RW/NO           X
mpu_show_regioninfo: 6                  0x0             0x20            DISABLED        RW/NO           X
mpu_show_regioninfo: 7                  0x0             0x20            DISABLED        RW/NO           X
mpu_show_regioninfo: ********************************************************************************************
elf_show_all_bin_section_addr: [app1] Text Addr : 0x217d040, Text Size : 8032
elf_show_all_bin_section_addr: [app2] Text Addr : 0x21fe4c0, Text Size : 364064
up_assert: Checking kernel heap for corruption...
up_assert: No kernel heap corruption detected
up_assert: Checking current app heap for corruption...
up_assert: No app heap corruption detected
up_assert: Assert location (PC) : 0x0e008fd8
up_assert: *******************************************
up_assert: Asserted TCB Info
up_assert: *******************************************
task_show_tcbinfo: State       : 3
task_show_tcbinfo: Flags       : 32
task_show_tcbinfo: Lock count  : 0
task_show_tcbinfo: Timeslice   : 10
task_show_tcbinfo: Waitdog     : 0
task_show_tcbinfo: WaitSem     : 0
task_show_tcbinfo: MsgwaitQ    : 0
task_show_tcbinfo: Sigdeliver  : 0
task_show_tcbinfo: Nsyscalls   : 0
task_show_tcbinfo: Syscall     : 0x10072198
```
up_usagefault: #########################################################################
up_usagefault: PANIC!!! Usagefault at instruction: 0x0e002284
up_usagefault: FAULT TYPE: UNDEFINSTR (Undefined instruction executed).
up_usagefault: FAULT REGS: CFAULTS: 0x00010400
up_usagefault: #########################################################################
up_assert: Assertion failed at file:armv8-m/up_usagefault.c line: 120 task: Idle Task
up_dumpstate: Code asserted in normal thread!
up_dumpstate: Current SP is User Thread Idle Task SP: 10004f0c
up_dumpstate: User stack:
up_dumpstate:   base: 10004ff8
up_dumpstate:   size: 00000400
up_dumpstate:   used: 0000029c
up_stackdump: 10004f00: xxxxxxxx xxxxxxxx xxxxxxxx 10004f0c 100294fc 0e000000 00000000 00000000
up_stackdump: 10004f20: 0e006927 00000003 0e00706d 0e0066a9 00000020 00010400 00010400 10004f84
up_stackdump: 10004f40: 1002c218 10004f84 00000000 0e000000 00000000 00000000 0e0091e9 1002c20c
up_stackdump: 10004f60: 1002a9f0 00000006 10012171 100293d0 00000000 100294fc 00003fe0 0e0696e0
up_stackdump: 10004f80: 10011e4f 10004fd8 00000000 100293d0 00000000 100294fc 00003fe0 0e0696e0
up_stackdump: 10004fa0: 0e000000 00000000 00000000 ffffffb8 00000000 00000000 0000002e 0e051c48
up_stackdump: 10004fc0: 48012000 00000000 0e002275 0e002281 0e002284 61000000 1005e070 10010f25
up_stackdump: 10004fe0: 100588e0 0001e720 deadbeef 00000000 deadbeef deadbeef deadbeef f0bf98ff
up_registerdump: R0: 0000002e 0e051c48 48012000 00000000 100293d0 00000000 100294fc 00003fe0
up_registerdump: R8: 0e0696e0 0e000000 00000000 00000000 0e002275 10004fd8 0e002281 0e002284
up_registerdump: xPSR: 61000000 BASEPRI: 00000000 CONTROL: 00000000
```
Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>
Co-authored-by: Kishore S N <kishore.sn@samsung.com>

thapav committed:
tools: Update TRAP tool based on latest assert logs

This patch updates the TRAP tool based on latest assert logs:
1) Adds IRQ parsing for assertion in IRQ mode.
2) Adds parsing for assertion in other modes (workqueue, normal, etc).

TRAP tool logs are as below:
*************************************************************
dump_file                   : None
log_file                    : latestlogs
Number of binaries          : 3 [kernel + apps]
"kernel" elf_file           : ../../build/output/bin/tinyara.axf
"app1" elf_file             : ../../build/output/bin/app1_dbg
"app2" elf_file             : ../../build/output/bin/app2_dbg
*************************************************************

-----------------------------------------------------------------------------------------
1. Crash Binary             : kernel

2. Crash type               : hard fault
   Crash log
	- up_assert: Assertion failed at file:armv8-m/up_hardfault.c line: 204 task: tash

3. Crash point (PC or LR)

	[ Caller - return address (LR) - of the function which has caused the crash ]
	- symbol addr        : 0x10012169
	- function name      : up_doirq
	- file               : /root/tizenrt/os/arch/arm/src/armv8-m/up_doirq.c:188

	[ Current location (PC) of assert ]
	- symbol addr        : 0x0e008fd8
	- function name      : up_memfault
	- file               : /root/tizenrt/os/arch/arm/src/armv8-m/up_memfault.c:120

	[ Exact crash point might be -4 or -8 bytes from the PC ]
	- symbol addr of (pc - 4)       : 0x0e008fd4
	- function name of (pc - 4)     : up_memfault
	- file of (pc - 4)              : /root/tizenrt/os/arch/arm/src/armv8-m/up_memfault.c:120

	[ Exact crash point might be -4 or -8 bytes from the PC ]
	- symbol addr of (pc - 8)       : 0x0e008fd0
	- function name of (pc - 8)     : up_interrupt_context
	- file of (pc - 8)              : /root/tizenrt/os/arch/arm/src/common/up_interruptcontext.c:87

	- Code asserted in IRQ state.

4. Assertion Data during interrupt mode:

- Interrupt handler at addr		Symbol_name
0xe008f75				/root/tizenrt/os/arch/arm/src/armv8-m/up_memfault.c:117

	- Interrupt number		: 4

5. Call stack of last run thread

	- Current stack pointer:		 0x218562c

Stack_address	 Symbol_address	 Symbol location  Symbol_name		File_name
 IRQ Stack
0x2185644	 0xe0069ec	 kernel  	  lowvsyslog	/root/tizenrt/os/include/syslog.h:263
0x218564c	 0xe00715c	 kernel  	  lowoutstream_putc	/root/tizenrt/lib/libc/stdio/lib_lowoutstream.c:76
0x2185650	 0xe00679c	 kernel  	  lib_noflush	/root/tizenrt/os/include/tinyara/streams.h:605
0x218567c	 0xe0093e4	 kernel  	  up_hardfault	/root/tizenrt/os/arch/arm/src/armv8-m/up_hardfault.c:119
0x2185680	 0xe0093e4	 kernel  	  up_hardfault	/root/tizenrt/os/arch/arm/src/armv8-m/up_hardfault.c:119
0x2185684	 0x10012144	 kernel  	  up_doirq	/root/tizenrt/os/arch/arm/src/armv8-m/up_doirq.c:92
0x218569c	 0x10011df8	 kernel  	  exception_common	/root/tizenrt/os/arch/arm/src/armv8-m/up_exception.S:163
0x21856e8	 0x10012144	 kernel  	  up_doirq	/root/tizenrt/os/arch/arm/src/armv8-m/up_doirq.c:92
0x21856ec	 0xe008fd4	 kernel  	  up_memfault	/root/tizenrt/os/arch/arm/src/armv8-m/up_memfault.c:117
0x21856f4	 0x10011df8	 kernel  	  exception_common	/root/tizenrt/os/arch/arm/src/armv8-m/up_exception.S:163
0x218570c	 0x10011df8	 kernel  	  exception_common	/root/tizenrt/os/arch/arm/src/armv8-m/up_exception.S:163
0x2185754	 0x10026fa4	 kernel  	  memcpy	/root/tizenrt/os/include/string.h:321
0x2185758	 0x1f40  	 app2    	  path.5669
0x218575c	 0x7099  	 app2    	  mpu_tc_main	/root/tizenrt/apps/examples/mpu/mpu_test.c:33
0x2185778	 0x3cb5  	 app2    	  tash_execute_cmd	/root/tizenrt/apps/shell/tash_command.c:538
0x21857a8	 0x4249  	 app2    	  tash_execute_cmdline	/root/tizenrt/apps/shell/tash_main.c:385
0x21857dc	 0xe001eb2	 kernel  	  sem_freeholder	/root/tizenrt/os/kernel/semaphore/sem_holder.c:163
0x21857e0	 0x3fb9  	 app2    	  tash_read_input_line	/root/tizenrt/apps/shell/tash_main.c:110
0x21857e4	 0x1e48  	 app2    	  g_pack_wifi_event.6230
0x21857f0	 0x3fb9  	 app2    	  tash_read_input_line	/root/tizenrt/apps/shell/tash_main.c:110
0x2185850	 0x4371  	 app2    	  tash_main	/root/tizenrt/apps/shell/tash_main.c:320
0x2185858	 0x4371  	 app2    	  tash_main	/root/tizenrt/apps/shell/tash_main.c:320
0x2185868	 0x1f78  	 app2    	  buffer.5651
0x2185870	 0xe001c48	 kernel  	  task_start	/root/tizenrt/os/kernel/task/task_start.c:133

x. Miscellaneous information:
-----------------------------------------------------------------------------------------

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

amogh-samsung committed:
os/fs/smartfs: Guaranteeing jSmartFs File Data Consistency during partial Overwrite

- Currently, jSmartFs guarantees only file system consistency, an overwrite
transaction is committed one sector at a time.
- This design does not ensure the presence of an entire file version, either
from before or after the modification.
- The new logical design commits the transaction for the entire file at once,
and hence ensures the presence of either the complete old or new version.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

amogh-samsung commented:
1. Ignore changes to the apps folder
2. Ignore comments added for debugging (fdbg)
3. TODO - release old sector chain after switching to new one
4. Implementation without CRC is not done yet and might not be required.

amogh-samsung commented:
Some simple tests work fine with this commit
Manual powercut test also ran fine, only tested for a few iterations. Auto powercut test will be done for 7-10 days

amogh-samsung commented:
Added logic to release old branch's chain sectors

ziyik committed:
os/board/rtl8720e: Add RTK Key Derive Function API (PBKDF2)

yeetee179 committed:
os/board/rtl/8720e: Add in BLE non_conn adv ability when slave link is already connected (#5691)

Taejun-Kwon commented on os/fs/smartfs/smartfs_smart.c:
```diff
@@ -255,6 +256,14 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 			}
 		}
 
+		for (ofile_ptr = fs->fs_head; ofile_ptr != NULL; ofile_ptr = ofile_ptr->fnext) {
```
it just implemented what I told you but it should allow if flag is read-only.
so when it commit changes, it should parse sf again and change offset & sector of sf which reading same entry.
Of course you can keep current status for trying but my comment need to be considered. there are many use-case write & read happened in different thread at the same time especially streaming case.

amogh-samsung replied:
All use cases where different threads/different fds open the same file are included here. 2 instances cannot be opened at the same time.
What is your idea of blocking second fd.
According to the durability principle, if one operation returns success, it's changes should be visible in the next query. But if we hold some data in the buffer, then the changes are not committed to flash. Hence I think blocking is fine till the file  is closed.
It is the responsibility of the app developer to make sure that the file is closed for syncing as soon as the operations are complete.

jiangxiaoyunjoy committed:
os/board/rtl8721csm: Modify do-while check in rtw_ble_server_start_adv(...) API V2 (#5693)

yeetee179 committed:
os/board/rtl/8720e: Fix BLE scan stop delay issue. (#5679)

Issue: BLE scan does not stop immediately, instead with a long delay.
Cause: Time is spent polling for register values when controller reports scan results to host.
Fix: Register reading is not necessary, thus removed.

sunghan-chang committed:
fix typos - recieve,fucntion,diable

recieve -> receive
fucntion -> function
diable -> disable

Signed-off-by: sunghan <sh924.chang@samsung.com>

sunghan-chang committed:
examples/board_specific_demo/ameba_*: fix typo of entry config

Kconfig_Entry should define a ENTRY_* config to provide entry point selection in menuconfig.
But ameba_secure_storage_demo and ameba_wifi_csi_demo duplicate define EXAMPLES_* configs
which are already defined in Kconfig.
This commit fixes the error on menuconfig as shown below.

/root/tizenrt/apps/examples/board_specific_demo/ameba_secure_storage_demo/Kconfig:7:warning: choice value used outside its choice group
/root/tizenrt/apps/examples/board_specific_demo/ameba_secure_storage_demo/Kconfig:8:warning: defaults for choice values not supported
/root/tizenrt/apps/examples/board_specific_demo/ameba_wifi_csi_demo/Kconfig:7:warning: choice value used outside its choice group
/root/tizenrt/apps/examples/board_specific_demo/ameba_wifi_csi_demo/Kconfig:8:warning: defaults for choice values not supported
/root/tizenrt/apps/apps_entry/Kconfig:8:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
/root/tizenrt/apps/apps_entry/Kconfig:8:	choice <choice> contains symbol EXAMPLES_SSTORAGE
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
/root/tizenrt/apps/examples/board_specific_demo/ameba_secure_storage_demo/Kconfig_ENTRY:1:	symbol EXAMPLES_SSTORAGE depends on EXAMPLES_SSTORAGE
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
/root/tizenrt/apps/examples/board_specific_demo/ameba_secure_storage_demo/Kconfig_ENTRY:1:	symbol EXAMPLES_SSTORAGE is part of choice <choice>

Signed-off-by: sunghan <sh924.chang@samsung.com>

sunghan-chang committed:
drivers/bluetooth/virtual_ble: remove duplicated config, VIRTUAL_BLE

The Kconfig under os/drivers/bluetooth defines the config VIRTUAL_BLE already.
But commit 3665952 duplicate defines it. That's why it makes the warning as below.
This commit fixes it.

drivers/bluetooth/virtual_ble/Kconfig:2:warning: choice value used outside its choice group
drivers/bluetooth/virtual_ble/Kconfig:3:warning: defaults for choice values not supported
drivers/bluetooth/Kconfig:27:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
drivers/bluetooth/Kconfig:27:	choice <choice> contains symbol VIRTUAL_BLE
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
drivers/bluetooth/Kconfig:31:	symbol VIRTUAL_BLE depends on VIRTUAL_BLE
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
drivers/bluetooth/Kconfig:31:	symbol VIRTUAL_BLE is part of choice <choice>

Signed-off-by: sunghan <sh924.chang@samsung.com>

CookieDoughMixer committed:
os/tools: terminate build when binary size exceeds partition size

When the binary size exceeds the partition size, the build should
terminate, but instead the build continues. Fixed this by exiting
the parent process.

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

kishore-sn committed:
os, lib: Fix Svace issues

lib_sscanf.c:700 - A string is copied to the array '&tmp[0]' of size 128, without checking its length first at lib_sscanf.c:700.
lib_strdup.c:73 - Use of vulnerable function 'strcpy' at lib_strdup.c:73. This function is unsafe, use strncpy instead.
bchdev_unregister.c:124 - Uninitialized data is read from local variable 'bch' at bchdev_unregister.c:124.
bchlib_write.c:139 - Division by zero
bchlib_read.c:152 - Division by zero
mm_check_heap_corruption.c:205 - After having been compared to NULL value at mm_check_heap_corruption.c:165,
pointer 'prev' is passed as 1st parameter in call to function 'dump_node' at mm_check_heap_corruption.c:205, where
it is dereferenced at mm_check_heap_corruption.c:104.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

sunghan-chang committed:
configs/rtl8720e/loadable_apps: fix typo, rtl8721csm

rtl8721csm -> rtl8720e

Signed-off-by: sunghan <sh924.chang@samsung.com>

sunghan-chang committed:
arch/arm/amebalite: add amebalite_signature.c in Make.defs

Because amebalite_signature.c file is ready, this commit
includes that to build in the chip.

Signed-off-by: sunghan <sh924.chang@samsung.com>

kaikaikaime committed:
Firmware upgrade for AmebaLite library

New partitions are added or modified to enhance clarity
Rephrasing of the variable for clearer distinction
Bug fixes in Wifi & peripherals reported by QC
Update img3 secure images

swasthik22 committed:
os/include/debug.h: Change the conditionals for mfdbg from dbg to lldbg

This patch is for change the mfdbg to use lldbg instead of dbg in flat build
The logs get stuck when there is heap currption
Change the mfdbg to use lldbg, when in abort mode or in interrupt context.
With the change the logs are getting displayed correctly.

WITH DBG -
mm_check_heap_corruption: ################################################
mm_check_heap_corruption: ERROR: Heap n ( logs are missing here)

WITH LLDBG -
mm_check_heap_corruption: ################################################
mm_check_heap_corruption: ERROR: Heap node corruption detected.
mm_check_heap_corruption: ================================================
mm_check_heap_corruption: Possible corruption scenario 1:
mm_check_heap_corruption: ...........

ziyik committed:
build/configs/rtl8720e/scripts: Correct PSRAM Heap size and layout
Current LD has removed PSRAM layout and assigned incorrect PSRAM HEAP start address
Add PSRAM layout into LD and correct PSRAM Heap start address.

ziyik committed:
build/configs/rtl8720e: Combine is and ns LD
Use common LD for both is and ns
Fix build error when Trustzone is not enable

swasthik22 committed:
arch/xtensa: add abort_mode variable to fix build break

declare abort_mode variable in xtensa_assert.c to fix the build break as it is used in common code.

kishore-sn committed:
os/mm: Print largest and total free size at allocation failure

When memory alloc is failed, we must always print largest free
node and total free space in the heap.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

Commit:
build/configs/rtl8720e: RAM optimization to reserve 3.8M heap
Change SRAM and PSRAM layout to reverse 3834KB heap for application. Reduce KR4_BD_PSRAM from 400k to 200k, and remove secure regions in SRAM.
kr4_image2_all.bin is built based on 10.0d release sdk. It has enabled BT configs and use "Code XIP Data Sram" ld config.

kishore-sn committed:
os/binfmt: Add code to free the allocated elf sections

The elf_addrenv_free() API is empty. Add code to free the
allocated elf sections. This API will be called when there is any
error during elf load and bind stage. So, if the elf sections have
already been allocated, they need to be freed here.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

yeetee179 committed:
os/board/rtl8721csm: BLE server disconnection fix (#5708)

- BLE server now disconnects based on the connection id

zhongnuo-tang committed:
os/board/rtl8720e/src/component/wifi/: Update Wificsi API

1.Insert CSI header information before CSI buffer, so no more reports of CSI header information via API
2.Add CSI buffer layout and CSI header information format
3.Add new parameters: csi_signature=0xABCD, hdr_len, csi_sequence in CSI header information
4.Modify variable name: from csi_raw_data_len to csi_data_len in CSI example code

yeetee179 committed:
os/board/rtl8721csm: BLE read, write and indication semaphore changes (#5700)

- Indication will always give semaphore when it finishes
- If slave has disconnected, give the indication semaphore
- When obtained, semapore will be deleted in rtw_ble_client_operation_read/write and rtw_ble_server_charact_indicate

thapav committed:
dbuild.sh: Refactor docker script

1) Rename current docker options to:
======================================================
  "1. Build with Current Configuration"
  "2. Re-configure"
  "3. Modify Current Configuration"
  "4. Clean Build"
  "5. Clean Build and Re-Configure"
  "6. Build SmartFS Image"
  "d. Download"
  "t. Build Test"
  "x. Exit"
======================================================
2) Fix typo in the script excute > execute
3) Fix typo in the script treate > treat

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

swasthik22 committed:
mm_check_heap_corruption.c : update the conditionals for the use of abort mode

This PR is for addressing the comments in PR #5705.
The abort mode flag is used in case of board crash. This flag is used in the mfdbg
definition in os/include/debug.h for mfdbg conditionals. So changing the conditional
for assignment of the board status for printing the heap information.

kishore-sn committed:
os/dutils.sh: Implement method to run toolchain commands in docker

When user has not installed toolchain, but only using the docker container,
then there needs to be a method to run toolchain commands like nm, readelf,
addr2line for debugging purpose.

This commit adds a method to run these commands. The user can call dutils.sh
with new "tool" command, followed by the actual toolchain command to be run.

./dutils.sh tool nm build/output/bin/tinyara.axf
./dutils.sh tool addr2line -e build/output/bin/tinyara.axf 0x0123abcd

The toolchaing commands can also be run from the dutils menu as below:

./dutils.sh menu
======================================================
  "Select Option"
======================================================
  "1. Run Toolchain Command"
  "x. Exit"
======================================================
1
make: Entering directory '/home/kishore/tinyara/TizenRT_kishore/os/tools'
chmod 755 ../makeinfo.sh
make: Leaving directory '/home/kishore/tinyara/TizenRT_kishore/os/tools'
======================================================
  "Select Binary for Toolchain Command"
======================================================
  "1. Kernel"
  "2. App1"
  "3. App2"
  "4. Common"
  "5. Enter manually"
  "x. Exit"
======================================================
1
======================================================
  "Select Toolchain Command"
======================================================
  "1. nm"
  "2. addr2line"
  "3. Enter manually"
  "x. Exit"
======================================================
2
Enter address:
e00d1fd
arm-none-eabi-addr2line -e ../build/output/bin/tinyara.axf e00d1fd
/root/tizenrt/os/drivers/serial/serial.c:1137
======================================================
  "Select Option"
======================================================
  "1. Run Toolchain Command"
  "x. Exit"
======================================================
x

Signed-off-by: Kishore S N <kishore.sn@samsung.com>
